server/stream.go

```diff
@@ -318,10 +318,14 @@ func (a *Account) addStreamWithAssignment(config *StreamConfig, fsConfig *FileSt
 		stype:     cfg.Storage,
 		consumers: make(map[string]*consumer),
 		msgs:      &inbound{mch: make(chan struct{}, 1)},
-		rmch:      make(chan uint64, 8192),
 		qch:       make(chan struct{}),
 	}
 
+	// For no-ack consumers when we are interest retention.
+	if cfg.Retention != LimitsPolicy {
```

Comment:
Can retention policy be changed/updated? If not good, otherwise, if user starts with different policy, then this channel won't be created and then later if policy is changed, this may cause problems. Unless this function itself would be invoked as part of a stream update?

Reply:
RetentionPolicy can not be updated at this time.

Reply:
I could add a nil check though when we place it, will do that.